### PR TITLE
[BPK-900] Use node 8.6.0 and npm@5

### PR DESCRIPTION
**Pull request by matthewdavidson**

### CODE REVIEW

**Comment by k0nserv** [comment]
Should we have package locks in our packages? Seems to me like we should ignore those and prevent them from being generated

**Comment by matthewdavidson** [comment]
@k0nserv `package-lock.json` is ignored from a library point of view. I originally thought the same as you but then i read it somewhere ... now to find the link 🤔

File: contributing.md [hunk 1]
```diff
@@ -75,11 +75,10 @@ When you are ready to jump into code, use the following workflow to add new comp
...
 
-> Make sure to run at least npm 4, which ships with Node 7. You can get the
-[latest Node version from this website](https://nodejs.org/en/), use a tool like
-[nvm](https://github.com/creationix/nvm) (`nvm use` will switch you to a supported version) or
-[nave](https://github.com/isaacs/nave) (`nave auto` does the trick), or just upgrade npm using
-`npm install -g npm@latest`.
+> Backpack development requires Node `8.6.0` and npm `5.3.0`. [Nvm](https://github.com/creationix/nvm) users can run
+`nvm use` to switch to `8.6.0`. [Nave](https://github.com/isaacs/nave) users can use `nave auto`. You can also
+download Node `8.6.0` using [their website](https://nodejs.org/en/). npm `5.3.0` comes with Node `8.6.0` by default. To
+install npm `5.3.0` manually, run `npm install --global npm@5.3.0`.
```
Comment (matteo-hertel) [inline contributing.md]: This should be  5.4.2 as in the `before_install` command right?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 node_js:
-  - 7.10.1
+  - 8.6.0
```
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply .travis.yml]: That seems to go against the notion of reproducible CI, but open to the idea at least.

If we want that then we could use `lts/*` syntax (see https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Specifying-Node.js-versions).

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 env:
   global:
     - NPM_CONFIG_LOGLEVEL=warn
+before_install:
+  - npm install -g npm@5.4.2
```
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply .travis.yml]: `npm install` will fail if npm@5.4.2 isn't installed. Consumers have to makes sure they install the correct version. The instructions are in the contrubuting.md

File: package.json [hunk 4]
```diff
@@ -1,11 +1,10 @@
 {
   "name": "backpack",
-  "engines": {
```
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply package.json]: I think it doesn't really provide value and is mostly meta data for libraries anyway.

Our preinstall checks do the same thing.

File: contributing.md (hunk 1 above)
Comment (matthewdavidson) [reply contributing.md]: oopsies 😇

File: package.json [hunk 5]
```diff
@@ -1,11 +1,10 @@
...
-    "node": ">=6.0.0",
-    "npm": ">=4.0.0"
-  },
   "private": true,
   "scripts": {
+    "preinstall:node": "node --version | grep -q 8.6.0",
+    "preinstall:npm": "npm --version | grep -q 5.4.2",
```
Comment (matteo-hertel) [inline package.json]: I'm assuming this will return exit code 1 i grep doesn't find the correct version, but will that stop the install?

Comment (k0nserv) [reply package.json]: Just verified with `"preinstall": "false"` and yes it does halt

Comment (matteo-hertel) [reply package.json]: Awesome, I wasn't 100% sure, thanks!

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply .travis.yml]: That's a good point

File: .travis.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply .travis.yml]: Looks good

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply package.json]: Is there any harm in keeping it?

[the remark this answers is not in the document]
Comment (matthewdavidson) [reply package.json]: I guess not haha 🙄

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply .travis.yml]: Do you mind if we wait until LTS lands at end of this month before I change everything to use `lts/*`?

[the remark this answers is not in the document]
Comment (k0nserv) [reply .travis.yml]: Nah I think keep it as is with hardcoded version and we'll make sure to update it with new LTS minors

**Pull request merged**